### PR TITLE
[FIX] base_vat: handle traceback while enter 'Tax ID' as  'FR'

### PR DESCRIPTION
**Pull request by mky-odoo**
When select 'France' into 'country' and enter 'FR' into 'Tax ID' save the record it will generate trackback.

Steps to produce:
 1. Into Invoice settings select 'verify Vat Number'
 2. Go to customer create new record
 3. Select country as a 'France' and enter Tax ID 'FR'
 
```
KeyError: ('res.partner', <function ResPartner._check_vies at 0x7f8b4bb0df30>, 'FR')
  File "odoo/tools/cache.py", line 91, in lookup
    r = d[key]
  File "<decorator-gen-3>", line 2, in __getitem__
  File "odoo/tools/func.py", line 87, in locked
    return func(inst, *args, **kwargs)
  File "odoo/tools/lru.py", line 34, in __getitem__
    a = self.d[obj]
Fault: INVALID_INPUT
  File "addons/base_vat/models/res_partner.py", line 128, in vies_vat_check
    vies_result = self._check_vies(country_code.upper() + vat_number)
  File "<decorator-gen-286>", line 2, in _check_vies
  File "odoo/tools/cache.py", line 96, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "addons/base_vat/models/res_partner.py", line 121, in _check_vies
    return check_vies(vat)
  File "stdnum/eu/vat.py", line 132, in check_vies
    return client.checkVat(number[:2], number[2:])
  File "zeep/proxy.py", line 46, in __call__
    return self._proxy._binding.send(
  File "zeep/wsdl/bindings/soap.py", line 135, in send
    return self.process_reply(client, operation_obj, response)
  File "zeep/wsdl/bindings/soap.py", line 229, in process_reply
    return self.process_error(doc, operation)
  File "zeep/wsdl/bindings/soap.py", line 329, in process_error
    raise Fault(
```

Applying this changes will resolve this issue.

Sentry-4209298700

### CODE REVIEW

**Comment by mky-odoo** [comment]
Hello @odoo/sentry_reviewers 
Please review this PR

Thanks

**Comment by abz-odoo** [comment]
@odoo/rd-accounting

**Comment by william-andre** [comment]
> `raise Fault(`...?

What is the message of the `Fault`?

**Comment by william-andre** [comment]
Then I think it deserves to be in the commit message or the PR description...

Also, on the sentry platform it is about a timeout.
Also, I can't reproduce with your steps
image

Also, why 16.0? Isn't that code the same in earlier versions?

**Comment by mky-odoo** [comment]
Hello @william-andre 
Here is video link please check it. https://www.awesomescreenshot.com/video/18137778?key=88c37980c986d50cb6c7f16e3d221fb1

**Comment by myhy-odoo** [comment]
> Hello @myhy-odoo This is the desired behavior.
> 
> `FR` isn't a valid input and thus, an error is thrown to the user telling "This isn't a valid FR VAT Number. A correct looks like FR...".
> 
> So, either the user input a correct VAT number or doesn't input a number.

Was I incorrectly tagged? I don't remember working on this issue.

**Comment by william-andre** [comment]
@mky-odoo @myhy-odoo obvious typo...

File: addons/base_vat/models/res_partner.py [hunk 1]
```diff
@@ -112,15 +113,18 @@ def simple_vat_check(self, country_code, vat_number):
...
     def vies_vat_check(self, country_code, vat_number):
         try:
             # Validate against  VAT Information Exchange System (VIES)
             # see also http://ec.europa.eu/taxation_customs/vies/
             vies_result = self._check_vies(country_code.upper() + vat_number)
-            return vies_result['valid']
+            return vies_result and vies_result['valid']
         except InvalidComponent:
             return False
         except Exception:
```
Comment (william-andre) [inline addons/base_vat/models/res_partner.py:130]: What you might want to do is set `sentry_ignored` here for the `_logger.exception`
Not sure if it works @yelizariev ?

**Pull request closed without merging**